guides/common/modules/proc_creating-a-custom-certificate.adoc

```diff
@@ -0,0 +1,106 @@
+[id="creating-a-custom-certificate_{context}"]
+
+= Creating a Custom SSL Certificate for {ProductName}
+
+ifeval::["{context}" == "{project-context}"]
+Use this procedure to create a custom SSL certificate for {ProductName}.
+If you already have a custom SSL certificate for {ProductName}, skip this procedure.
+endif::[]
+
+ifeval::["{context}" == "{smart-proxy-context}"]
+On {ProjectServer}, create a custom certificate for your {ProductName}.
+If you already have a custom SSL certificate for {ProductName}, skip this procedure.
+endif::[]
+
+.Procedure
+
+. To store all the source certificate files, create a directory that is accessible only to the `root` user.
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# mkdir /root/{context}_cert
+----
+
+. Create a private key with which to sign the Certificate Signing Request (CSR).
++
+Note that the private key must be unencrypted.
+If you use a password-protected private key, remove the private key password.
++
+If you already have a private key for this {ProductName}, skip this step.
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# openssl genrsa -out `/root/{context}_cert/{context}_cert_key.pem` 4096
+----
+
+. Create the `/root/{context}_cert/openssl.cnf` configuration file for the Certificate Signing Request (CSR) and include the following content:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+[ req ]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+x509_extensions = usr_cert
+prompt = no
+
+[ req_distinguished_name ] <1>
+C  = _Country Name (2 letter code)_
+ST = _State or Province Name (full name)_
+L  = _Locality Name (eg, city)_
+O  = _Organization Name (eg, company)_
+OU = _The division of your organization handling the certificate_
+CN = _{context}.example.com_ <2>
+
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+extendedKeyUsage = serverAuth, clientAuth, codeSigning, emailProtection
+subjectAltName = @alt_names
+
+[ usr_cert ]
+basicConstraints=CA:FALSE
+nsCertType = client, server, email
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth, clientAuth, codeSigning, emailProtection
+nsComment = "OpenSSL Generated Certificate"
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid,issuer
+
+[ alt_names ]
+DNS.1 = _{context}.example.com_ <3>
+----
+ifeval::["{context}" == "{project-context}"]
+<1> In the `[ req_distinguished_name ]` section, enter information about your organization.
```

Comment:
I'd put this at the end and mark it as optional, or omit it.

Reply:
Omitted :+1: